tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+```
+
+
+## Solution: Collection of related Projects
+
+The section [Project setup for related projects](#project-setup-for-related-projects) describes the collection of related projects.  The file `*.csolution.yml` describes the relationship of this projects.  This file may also define [Target and Build Types](#target-and-build-types) before the section `solution:`.
+
+The YML structure of the section `solution:` is:
+
+```yml
+solution:                  # Start a list of projects.
+  - project:               # path to the project file (required).
+    for-type:              # include project for a list of *build* and *target* types (optional).
+    not-for-type:          # exclude project for a list of *build* and *target* types (optional).
+    compiler:              # specify a specific compiler
+    optimize:              # optimize level for code generation (optional)
+    debug:                 # generation of debug information (optional)
+    defines:               # define symbol settings for code generation (optional).
+    undefines:             # remove define symbol settings for code generation (optional).
+    add-paths:             # additional include file paths (optional).
+    del-paths:             # remove specific include file paths (optional). 
+    misc:                  # Literal tool-specific controls
+```
+
+## Project Definition
+
+Start of a project definition in a `*.cproject.yml` file (optional section)
+
+The YML structure of the section `project:` is:
+```yml
+project:                     # do we need this section at all, perhaps for platform?
+  compiler: name             # specify compiler (AC6, GCC, IAR)
+  description:               # project description (optional)
+  output-type: lib | exe     # generate executable (default) or library
+  processor:                 # specify processor attributes 
+  groups:                    # source files organized in groups
+  layers:                    # software layers of pre-configured software components
+  components:                # software components
+```  
+
+
+## Processor Attributes
+
+The `processor:` keyword defines attributes such as TrustZone and FPU register usage.
+
+```yml
+ processor:                # processor specific settings
+    trustzone: secure      # TrustZone mode: secure | non-secure | off
+    fpu: on | off          # control usage of FPU registers (S-Register for Helium/FPU) (default: on for devices with FPU registers)
+    endian: little | big   # select endianess (only available when devices are configureable)
+```
+
+**Note:**
+- Default for `trustzone:` 
+   - `off` for devices that support this option.
+   - `non-secure` for devices that have TrustZone enabled.
+   
+
+# Pre/Post build steps
+
+Tbd: potentially map to CMake add_custom_command.
+
+```yml
+- execute: description      # execute an external command with description
+  os: Linux                 # executed on which operating systems (if omitted it is OS independent)
```

Comment:
I think that the same pre or post build step should have a "for-os" directive so that we can invoke different tools depending on the OS.